download_slashing_interchange_tests.sh

```diff
@@ -29,7 +36,7 @@ dl_version() {
 	for flavour in "${FLAVOURS[@]}"; do
 		if [[ ! -e "${flavour}.tar.gz" ]]; then
 			echo "Downloading: slashing-${version}/${flavour}.tar.gz"
-			curl --location --remote-name --show-error \
+			curl --location --remote-name --silent --show-error --retry 3 --retry-all-errors \
```

Comment:
I had noticed that after I tried to bump this submodule in `nimbus-eth2` the other day, it didn't work on at least some systems, and now I think this sort of change is why. https://github.com/status-im/nim-eth2-scenarios/pull/10 does this as well, and while the `--retry 3` should be fine, https://daniel.haxx.se/blog/2020/06/24/curl-7-71-0-blobs-and-retries/ documents that `--retry-all-errors` only appeared post-Ubuntu 18.04 and 20.04 LTS releases. One could argue if it were only the former that it's going EOL soon, and better to have `--retry-all-errors` than continue to support 18.04 the final few months, but since both https://packages.ubuntu.com/bionic-updates/curl (18.04LTS) has 7.58 and even https://packages.ubuntu.com/focal-updates/curl (20.04LTS) has 7.68, this eminently useful `--retry-all-errors` flag isn't available.

So not only probably can it not be used here, but https://github.com/status-im/nim-eth2-scenarios/pull/10 probably has to be reverted.

https://curl.se/docs/manpage.html#--retry-all-errors documents some other ways of getting some other error handling, e.g., `--retry-connrefused` appeared in 7.52.0 so it's safe, so it might be possible to reconstruct most of the useful parts while remaining 20.04LTS and even 18.04LTS-compatible.

Does macOS update `curl` and if so, what versions does it have, e.g., for macOS 10.15? It shouldn't evoke the GPLv3 concerns of GNU coreutils/bash/gcc/etc, so there's been no equivalently obvious reason for Apple to have stopped updating it. In particular, Big Sur should be supported until it reaches EOL presumably in about a year.

Reply:
```
% sw_vers
ProductName:		macOS
ProductVersion:		13.0.1
BuildVersion:		22A400
% curl --version
curl 7.84.0 (x86_64-apple-darwin22.0) libcurl/7.84.0 (SecureTransport) LibreSSL/3.3.6 zlib/1.2.11 nghttp2/1.47.0
Release-Date: 2022-06-27
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL threadsafe UnixSockets
```

Reply:
Replaced with the more portable `--retry-connrefused` for now. 
if we still see curl failing occasionally, maybe an option could be to build the retry loop around it manually.

Reply:
From Jenkins (x86_64):

```
+ sw_vers
17:40:59  ProductName:	macOS
17:40:59  ProductVersion:	12.3.1
17:40:59  BuildVersion:	21E258
[Pipeline] sh
17:40:59  + curl --version
17:40:59  curl 7.79.1 (x86_64-apple-darwin21.0) libcurl/7.79.1 (SecureTransport) LibreSSL/3.3.5 zlib/1.2.11 nghttp2/1.45.1
17:40:59  Release-Date: 2021-09-22
17:40:59  Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
17:40:59  Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets
[Pipeline]
```

Jenkins (aarch64)

```
17:40:59  + sw_vers
17:40:59  ProductName:	macOS
17:40:59  ProductVersion:	12.3.1
17:40:59  BuildVersion:	21E258
[Pipeline] sh
17:41:00  + curl --version
17:41:00  curl 7.79.1 (x86_64-apple-darwin21.0) libcurl/7.79.1 (SecureTransport) LibreSSL/3.3.5 zlib/1.2.11 nghttp2/1.45.1
17:41:00  Release-Date: 2021-09-22
17:41:00  Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
17:41:00  Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets
```

Reply:
> From Jenkins (x86_64):
> 
> ```
> + sw_vers
> 17:40:59  ProductName:	macOS
> 17:40:59  ProductVersion:	12.3.1
> 17:40:59  BuildVersion:	21E258
> [Pipeline] sh
> 17:40:59  + curl --version
> 17:40:59  curl 7.79.1 (x86_64-apple-darwin21.0) libcurl/7.79.1 (SecureTransport) LibreSSL/3.3.5 zlib/1.2.11 nghttp2/1.45.1
> 17:40:59  Release-Date: 2021-09-22
> 17:40:59  Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
> 17:40:59  Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets
> [Pipeline]
> ```
> 
> Jenkins (aarch64)
> 
> ```
> 17:40:59  + sw_vers
> 17:40:59  ProductName:	macOS
> 17:40:59  ProductVersion:	12.3.1
> 17:40:59  BuildVersion:	21E258
> [Pipeline] sh
> 17:41:00  + curl --version
> 17:41:00  curl 7.79.1 (x86_64-apple-darwin21.0) libcurl/7.79.1 (SecureTransport) LibreSSL/3.3.5 zlib/1.2.11 nghttp2/1.45.1
> 17:41:00  Release-Date: 2021-09-22
> 17:41:00  Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
> 17:41:00  Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets
> ```

Alright, seems like Ubuntu is the bottleneck here.

Reply:
Should I split the curl arg update into separate PR or alright as submarine in here?

Reply:
> Replaced with the more portable `--retry-connrefused` for now. if we still see curl failing occasionally, maybe an option could be to build the retry loop around it manually.

Agree, still useful to reduce spurious CI failures, and would rather not wait for 2025, when 20.04LTS should lose support, for that to happen. Unaesthetic but likely at some point necessary.

Reply:
> Should I split the curl arg update into separate PR or alright as submarine in here?

Fine here